index.js

```diff
@@ -127,20 +133,22 @@ export class StuckTransactionsCanceller {
         log: str => this.#log(str),
         sendTransaction: tx => this.#sendTransaction(tx)
       })
+      await this.addPending(replacementTx)
     } catch (err) {
       if (err.code === 'NONCE_EXPIRED') {
         this.#log(`${tx.hash} has already been confirmed`)
-        await this.#store.remove(tx.hash)
+        await this.removeConfirmed(tx)
         return
       } else {
         throw err
       }
     }
-    this.#log(`
-      Waiting for receipt of replacing ${tx.hash} with ${replacementTx.hash}...`
+    this.#log(
+      `Waiting for receipt of replacing ${tx.hash} with ${replacementTx.hash}...`
     )
     await replacementTx.wait()
-    await this.#store.remove(tx.hash)
+    await this.removeConfirmed(replacementTx)
+    await this.removeConfirmed(tx)
```

Comment:
Since `tx.nonce === replacementTx.nonce` and `removeConfirmed` removes all transactions with the same nonce, I think it is safe to call `removeConfirmed` only once.

WDYT?

Reply:
Dang it, forget to push a commit 😅 Already fixed